src/main/java/randoop/generation/DemandDrivenInputCreation.java

```diff
@@ -0,0 +1,489 @@
+            if (executable instanceof Method && !Modifier.isStatic(executable.getModifiers())) {
+              inputTypeList.add(0, new NonParameterizedType(currentClass));
+            }
+            TypeTuple inputTypes = new TypeTuple(inputTypeList);
+            CallableOperation callableOperation =
+                executable instanceof Constructor
+                    ? new ConstructorCall((Constructor<?>) executable)
+                    : new MethodCall((Method) executable);
+            NonParameterizedType declaringType = new NonParameterizedType(currentClass);
+            TypedOperation typedClassOperation =
+                new TypedClassOperation(callableOperation, declaringType, inputTypes, returnType);
+
+            // Add the method call to the producerMethods.
+            producerMethodsList.add(typedClassOperation);
+            producerParameterTypes.addAll(inputTypeList);
+          }
+          processed.add(currentType);
+          // Recursively search for methods that construct objects of the specified type.
+          workList.addAll(producerParameterTypes);
+        }
+      }
+      initialCall = false;
+    }
+
+    // TODO: Reverse the producerMethodsList may improve the quality of the generated tests.
+    // Producer methods are added to the list in the order they are needed. However, objects are
+    // often built up from the simplest types. Reversing the producerMethodsList may help generate
+    // basic types first, leading to the generation of more complex types within fewer tests.
+    // This needs to be looked into further.
+    Collections.reverse(producerMethodsList);
+
+    Set<TypedOperation> producerMethods = new LinkedHashSet<>(producerMethodsList);
+
+    return producerMethods;
+  }
+
+  /**
+   * Given an array of classes, this method converts them into a list of Types.
+   *
+   * @param classes an array of reflection classes
+   * @return a list of Types
+   */
+  private static List<Type> classArrayToTypeList(Class<?>[] classes) {
+    return CollectionsPlume.mapList(Type::forClass, classes);
+  }
+
+  /**
+   * Given a TypedOperation, this method finds a sequence of method calls that can generate an
+   * instance of each input type required by the TypedOperation. It then merges these sequences into
+   * a single sequence.
+   *
+   * @param sequenceCollection the SequenceCollection from which to draw input sequences
+   * @param typedOperation the operation for which input sequences are to be generated
+   * @return a sequence that ends with a call to the provided TypedOperation, or null if no such
+   *     sequence can be found
+   */
+  private static @Nullable Sequence getInputAndGenSeq(
+      SequenceCollection sequenceCollection, TypedOperation typedOperation) {
+    TypeTuple inputTypes = typedOperation.getInputTypes();
+    List<Sequence> inputSequences = new ArrayList<>();
+
+    // Represents the position of a statement in a sequence.
+    int index = 0;
```

Comment:
The comment states the type of the variable (index into a sequence), but not its semantics:  what does it mean or how is it used?